src/V3Timing.cpp

```diff
@@ -0,0 +1,785 @@
+            m_timingKit.postStmts
+                = VN_AS(AstNode::addNext(m_timingKit.postStmts, new AstCCall{flp, m_commitFuncp}),
+                        NodeStmt);
+        }
+        m_commitFuncp->addStmtsp(stmtsp);
+    }
+    // Creates a new active with the given statements under always
+    void createActive(AstSenTree* const sensesp, AstNodeStmt* stmtsp = nullptr) {
+        auto* const flp = sensesp->fileline();
+        auto* const activep = new AstActive{flp, "", sensesp};
+        m_scopeTopp->addActivep(activep);
+        activep->addStmtsp(new AstAlways{flp, VAlwaysKwd::ALWAYS, nullptr, stmtsp});
+    }
+    // Creates a trigger scheduler variable together with a commit() call
+    AstVarScope* getCreateSchedulerp(AstSenTree* const sensesp) {
+        if (!sensesp->user1()) {
+            auto* const trigSchedp
+                = m_scopeTopp->createTemp(m_trigSchedNames.get(sensesp), m_trigSchedDtp);
+            auto* const flp = sensesp->fileline();
+            sensesp->user1p(trigSchedp);
+            // Create commit() and put it in the global commit function
+            auto* const commitp = new AstCMethodHard{
+                flp, new AstVarRef{flp, trigSchedp, VAccess::WRITE}, "commit"};
+            commitp->statement(true);
+            commitp->dtypeSetVoid();
+            addToCommitFunc(commitp);
+            // Create resume() and put it in a new active
+            auto* const resumep = new AstCMethodHard{
+                flp, new AstVarRef{flp, trigSchedp, VAccess::WRITE}, "resume"};
+            resumep->statement(true);
+            resumep->dtypeSetVoid();
+            createActive(sensesp, resumep);
+            return trigSchedp;
+        }
+        return VN_AS(sensesp->user1p(), VarScope);
+    }
+    // Creates a comment describing the sentree
+    AstComment* createComment(AstSenTree* const sensesp) {
+        if (!sensesp->user2()) {
+            std::stringstream ss;
+            V3EmitV::verilogForTree(sensesp, ss);
+            auto* const commentp = new AstComment{sensesp->fileline(), ss.str()};
+            sensesp->user2p(commentp);
+            return commentp;
+        }
+        return VN_AS(sensesp->user2p(), Comment)->cloneTree(false);
+    }
+
+    // VISITORS
+    virtual void visit(AstScope* nodep) override {
+        VL_RESTORER(m_scopep);
+        m_scopep = nodep;
+        iterateChildren(nodep);
+    }
+    virtual void visit(AstFork* nodep) override {
+        auto* const flp = nodep->fileline();
+
+        // Create unique name for this fork
+        auto forkName = m_forkNames.get(nodep);
+        if (!nodep->unnamed()) forkName += nodep->name() + "__";
+
+        // Create a fork var if it's not join_none
+        AstVarScope* forkVscp = nullptr;
+        if (!nodep->joinType().joinNone()) {
+            auto* const forkVarp
+                = new AstVar{flp, VVarType::BLOCKTEMP, forkName + "__handle", m_forkDtp};
+            forkVarp->funcLocal(true);
```

Comment:
If you add this under the `m_scopep`, then it's not really a `funcLocal`. Not sure why this is here.